### PR TITLE
Mobilising programme

### PR DESCRIPTION
**Pull request by naazy**
mobilising prog (I opened PR but it was a joined effort by me and @harrygfox)

### CODE REVIEW

**Comment by naazy** [comment]
Looks great @harrygfox :smile:

File: src/MainModel.elm [hunk 1]
```diff
@@ -76,16 +78,30 @@ type SessionColumn
...
+        [ { id = 1, name = "Columnus Maximus" }
+        , { id = 2, name = "Columnus Minimus" }
+        ]
+    , locations =
+        [ { id = 3, name = "Focus Hub" }
+        , { id = 2, name = "Harry's House" }
+        , { id = 1, name = "Naaz's House" }
+        ]
+    , chairs =
+        [ { id = 2, name = "Harry's Mum" }
```
Comment (roryc89) [inline src/MainModel.elm]: If our api endpoint fails this will be what the user sees 😆

File: src/MainModel.elm [hunk 2]
```diff
@@ -105,13 +121,86 @@ initialModel =
...
+                , month = 10
+                , year = 2017
+                }
+          , sessions =
+                [ { chairId = Just 2
+                  , description = ""
+                  , endTime = { hour = 16, minute = 0 }
+                  , id = 2
+                  , locationId = Just 2
+                  , name = "Resting off the bevs"
```
Comment (roryc89) [inline src/MainModel.elm]: same here

File: src/MobileView.elm [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                                [ span [ class "prog-mob__day" ]
+                                    [ text ((DateUtils.dateWithoutTimeToDay d.date))
+                                    ]
+                                , span [ class "prog-mob__date" ] [ text (toString d.date.day ++ " " ++ (DateUtils.intToMonthString d.date.month)) ]
+                                ]
+                            , div []
+                                (List.map
+                                    (\s ->
+                                        let
+                                            locationId =
```
Comment (roryc89) [inline src/MobileView.elm]: we can use Maybe.withDefault here and below

File: src/MobileView.elm [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+                                            chairId =
+                                                case s.chairId of
+                                                    Just chairId ->
+                                                        chairId
+
+                                                    Nothing ->
+                                                        0
+
+                                            -- DateUtils.timeOfDayToTime dateWithSessions.date s.startTime
```
Comment (roryc89) [inline src/MobileView.elm]: left comment

File: src/MobileView.elm [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+                (\d ->
+                    let
+                        sortedSessions =
+                            d.sessions
+                                |> List.sortBy (\s -> DateUtils.timeOfDayToTime d.date s.startTime)
+                    in
+                        div [ class "prog-mob__oneday" ]
+                            [ div [ class "prog-mob__datecontainer" ]
+                                [ span [ class "prog-mob__day" ]
+                                    [ text ((DateUtils.dateWithoutTimeToDay d.date))
```
Comment (roryc89) [inline src/MobileView.elm]: no need for double parens

File: src/MainModel.elm [hunk 6]
```diff
@@ -42,6 +42,8 @@ type alias Model =
     , invalidSubmissionIdsInput : String
     , showValidation : Bool
     , scheduleSubmissionsIndividually : Bool
+    , showMobileView : Bool
+    , displayedColumns : List Column
```
Comment (roryc89) [inline src/MainModel.elm]: As far as I can tell we only use the id of the 1st column in this list. Do we need store a list of columns here or could it just be a `Maybe Int`? I understand if this due to planning for a future feature.

Comment (naazy) [reply src/MainModel.elm]: done

**Pull request merged**